panic: runtime error

Hello, I am pulling my hair out on this - trying to start a new cluster (no data in store).  As soon as the raft election occurs, one node crashes with:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x8049f3b]

Here is the full output, from start, with all debug enabled:

 $PWD/nats-streaming-server -sc $PWD/nss.2node.config
[11094] 2019/07/20 07:50:32.016304 [INF] STREAM: Starting nats-streaming-server[clus1] version 0.15.1
[11094] 2019/07/20 07:50:32.016612 [INF] STREAM: ServerID: 9NEjmlAD74YjT5QSUGD3Zc
[11094] 2019/07/20 07:50:32.016636 [INF] STREAM: Go version: go1.11.10
[11094] 2019/07/20 07:50:32.016655 [INF] STREAM: Git commit: [not set]
[11094] 2019/07/20 07:50:32.019050 [INF] Starting nats-server version 2.0.0
[11094] 2019/07/20 07:50:32.019117 [INF] Git commit [not set]
[11094] 2019/07/20 07:50:32.019966 [INF] Starting http monitor on 0.0.0.0:44227
[11094] 2019/07/20 07:50:32.020270 [INF] Listening for client connections on 0.0.0.0:4227
[11094] 2019/07/20 07:50:32.020328 [INF] Server id is NDP4KHELCOITHWNR437VLDV6OEABPKAH3V5BNEX5I6AMEK3JJCXZPZ6U
[11094] 2019/07/20 07:50:32.020351 [INF] Server is ready
[11094] 2019/07/20 07:50:32.021320 [INF] Listening for route connections on 0.0.0.0:6227
[11094] 2019/07/20 07:50:32.025759 [INF] 140.188.62.140:6227 - rid:1 - Route connection created
[11094] 2019/07/20 07:50:32.051774 [INF] STREAM: Recovering the state...
[11094] 2019/07/20 07:50:32.054273 [INF] STREAM: Recovered 0 channel(s)
[11094] 2019/07/20 07:50:32.054329 [INF] STREAM: Cluster Node ID : b
[11094] 2019/07/20 07:50:32.054373 [INF] STREAM: Cluster Log Path: /opt/NATS/cluster1
[11094] 2019/07/20 07:50:32.058389 [INF] STREAM:  raft: Initial configuration (index=1): [{Suffrage:Voter ID:b Address:clus1.b.clus1} {Suffrage:Voter ID:a Address:clus1.a.clus1}]
[11094] 2019/07/20 07:50:32.058457 [DBG] STREAM: Loaded existing state for Raft group clus1
[11094] 2019/07/20 07:50:32.058522 [DBG] STREAM: Discover subject:           _STAN.discover.clus1
[11094] 2019/07/20 07:50:32.058546 [DBG] STREAM: Publish subject:            _STAN.pub.clus1.>
[11094] 2019/07/20 07:50:32.058567 [DBG] STREAM: Subscribe subject:          _STAN.sub.clus1
[11094] 2019/07/20 07:50:32.058593 [DBG] STREAM: Subscription Close subject: _STAN.subclose.clus1
[11094] 2019/07/20 07:50:32.058614 [DBG] STREAM: Unsubscribe subject:        _STAN.unsub.clus1
[11094] 2019/07/20 07:50:32.058633 [DBG] STREAM: Close subject:              _STAN.close.clus1
[11094] 2019/07/20 07:50:32.058910 [INF] STREAM:  raft: Node at clus1.b.clus1 [Follower] entering Follower state (Leader: "")
[11094] 2019/07/20 07:50:32.059747 [INF] STREAM: Message store is RAFT_FILE
[11094] 2019/07/20 07:50:32.059785 [INF] STREAM: Store location: /opt/NATS/cluster1/store
[11094] 2019/07/20 07:50:32.059880 [INF] STREAM: ---------- Store Limits ----------
[11094] 2019/07/20 07:50:32.059902 [INF] STREAM: Channels:                  100 *
[11094] 2019/07/20 07:50:32.059922 [INF] STREAM: --------- Channels Limits --------
[11094] 2019/07/20 07:50:32.059940 [INF] STREAM:   Subscriptions:          1000 *
[11094] 2019/07/20 07:50:32.059958 [INF] STREAM:   Messages     :       1000000 *
[11094] 2019/07/20 07:50:32.059976 [INF] STREAM:   Bytes        :     976.56 MB *
[11094] 2019/07/20 07:50:32.059994 [INF] STREAM:   Age          :     unlimited *
[11094] 2019/07/20 07:50:32.060013 [INF] STREAM:   Inactivity   :     unlimited *
[11094] 2019/07/20 07:50:32.060031 [INF] STREAM: ----------------------------------
[11094] 2019/07/20 07:50:32.924810 [INF] 140.188.62.140:48830 - rid:7 - Route connection created
[11094] 2019/07/20 07:50:32.925166 [INF] 140.188.62.140:48830 - rid:7 - Router connection closed
[11094] 2019/07/20 07:50:33.589288 [TRC] STREAM: raft-net: clus1.b.clus1 accepted connection from: clus1.a.clus1
[11094] 2019/07/20 07:50:33.590075 [TRC] STREAM: raft: lost leadership because received a requestvote with newer term
[11094] 2019/07/20 07:50:33.590451 [WRN] STREAM:  raft: Rejecting vote request from clus1.a.clus1 since our last term is greater (260, 1)
[11094] 2019/07/20 07:50:35.568696 [WRN] STREAM:  raft: Heartbeat timeout from "" reached, starting election
[11094] 2019/07/20 07:50:35.568802 [INF] STREAM:  raft: Node at clus1.b.clus1 [Candidate] entering Candidate state in term 277
[11094] 2019/07/20 07:50:35.569495 [TRC] STREAM: raft: Votes needed: 2
[11094] 2019/07/20 07:50:35.569549 [TRC] STREAM: raft: Vote granted from b in term 277. Tally: 1
[11094] 2019/07/20 07:50:35.573495 [TRC] STREAM: raft: Vote granted from a in term 277. Tally: 2
[11094] 2019/07/20 07:50:35.573547 [INF] STREAM:  raft: Election won. Tally: 2
[11094] 2019/07/20 07:50:35.573608 [INF] STREAM:  raft: Node at clus1.b.clus1 [Leader] entering Leader state
[11094] 2019/07/20 07:50:35.573661 [INF] STREAM:  raft: Added peer a, starting replication
[11094] 2019/07/20 07:50:35.574299 [INF] STREAM: server became leader, performing leader promotion actions
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x8049f3b]

goroutine 88 [running]:
runtime/internal/atomic.Load64(0x9c8d3ac, 0x9ce4ea0, 0x4)
        /home/travis/.gimme/versions/go1.11.10.linux.amd64/src/runtime/internal/atomic/asm_386.s:188 +0xb
github.com/nats-io/nats-streaming-server/vendor/github.com/hashicorp/raft.(*Raft).replicateTo(0x9cb0000, 0x9c8d380, 0x12, 0x0, 0x0)
        /home/travis/gopath/src/github.com/nats-io/nats-streaming-server/vendor/github.com/hashicorp/raft/replication.go:205 +0xaf
github.com/nats-io/nats-streaming-server/vendor/github.com/hashicorp/raft.(*Raft).replicate(0x9cb0000, 0x9c8d380)
        /home/travis/gopath/src/github.com/nats-io/nats-streaming-server/vendor/github.com/hashicorp/raft/replication.go:154 +0x3bb
github.com/nats-io/nats-streaming-server/vendor/github.com/hashicorp/raft.(*Raft).startStopReplication.func1()
        /home/travis/gopath/src/github.com/nats-io/nats-streaming-server/vendor/github.com/hashicorp/raft/raft.go:488 +0x27
github.com/nats-io/nats-streaming-server/vendor/github.com/hashicorp/raft.(*raftState).goFunc.func1(0x9cb0000, 0x9ce9060)
        /home/travis/gopath/src/github.com/nats-io/nats-streaming-server/vendor/github.com/hashicorp/raft/state.go:146 +0x44
created by github.com/nats-io/nats-streaming-server/vendor/github.com/hashicorp/raft.(*raftState).goFunc
        /home/travis/gopath/src/github.com/nats-io/nats-streaming-server/vendor/github.com/hashicorp/raft/state.go:144 +0x54

Good day, some progress.  Our developer compiled a binary from the latest code, and that allowed me to get past the error I posted above.  However, now I am seeing an error when we try to use the service:

`[7975] 2019/07/22 12:11:55.865541 [DBG] STREAM: [Client:TestClient--Listener] Connected (Inbox=_INBOX.LCQWiY8jpCfBaLusOkj7AW)
[7975] 2019/07/22 12:11:55.870488 [INF] STREAM: Channel "Notifications" has been created
[7975] 2019/07/22 12:11:55.873554 [DBG] STREAM: [Client:TestClient--Listener] Started new subscription, subject=Notifications, inbox=_INBOX.LCQWiY8jpCfBaLusOkj7O4, subid=1, sending from sequence, asked_seq=1 actual_seq=1
[7975] 2019/07/22 12:11:55.881839 [DBG] STREAM: [Client:TestClient--Publisher] Connected (Inbox=_INBOX.LCQWiY8jpCfBaLusOkj7Uq)
panic: interface conversion: stores.SubStore is *stores.RaftSubStore, not *stores.FileSubStore

goroutine 1234 [running]:
github.com/nats-io/nats-streaming-server/stores.(*FileStore).autoSync(0xc000077380)
        /home/riddlmi/build_nats/nats-streaming-server-master/stores/filestore.go:1329 +0x33c
created by time.goFunc
        /usr/lib/golang/src/time/sleep.go:169 +0x44
`

We saw this after our crash, and thought it looked odd.  ("Message store is RAFT_FILE")
Is this just referencing the fact that raft messages are logging to file, not an indication that raft and our message streams are conflicting?


panic: interface conversion: stores.SubStore is *stores.RaftSubStore, not *stores.FileSubStore

goroutine 1556 [running]:
github.com/nats-io/nats-streaming-server/stores.(*FileStore).autoSync(0xc00010c300)
        /home/riddlmi/build_nats/nats-streaming-server-master/stores/filestore.go:1329 +0x33c
created by time.goFunc
        /usr/lib/golang/src/time/sleep.go:169 +0x44 
[20677] 2019/07/22 13:38:06.596739 [INF] STREAM: Cluster Node ID : b
[20677] 2019/07/22 13:38:06.596753 [INF] STREAM: Cluster Log Path: test-nss1/b
[20677] 2019/07/22 13:38:06.600464 [INF] STREAM: Message store is RAFT_FILE
[20677] 2019/07/22 13:38:06.600505 [INF] STREAM: Store location: store-nss1
[20677] 2019/07/22 13:38:06.600619 [INF] STREAM: ---------- Store Limits ----------

--- a/go.mod
+++ b/go.mod
@@ -6,7 +6,7 @@ require (
 	github.com/go-sql-driver/mysql v1.4.1
 	github.com/gogo/protobuf v1.2.1
 	github.com/hashicorp/go-msgpack v0.5.5
-	github.com/hashicorp/raft v1.1.0
+	github.com/hashicorp/raft v1.1.1
 	github.com/lib/pq v1.1.1
 	github.com/nats-io/nats-server/v2 v2.0.2
 	github.com/nats-io/nats.go v1.8.1

